src/components/ProductCard/Productcard.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import FavoriteBorderOutlinedIcon from "@mui/icons-material/FavoriteBorderOutlined";
+import FavoriteIcon from "@mui/icons-material/Favorite";
+import "./Productcard.css";
+import StarOutlinedIcon from "@mui/icons-material/StarOutlined";
+import { Link, useNavigate } from "react-router-dom";
+import { useData } from "../../contexts/Data-context";
+import { actionTypes } from "../../reducers/actionTypes";
+import { useAuth } from "../../contexts/Auth-context";
+import {
+  AddToCartService,
+  AddToWishlistService,
+  DeleteFromWishlistService,
+} from "../../services/apiCall";
+
+const Productcard = ({ product }) => {
+  const { token } = useAuth();
+  const { state, dispatch, loader, setLoader } = useData();
+  const navigate = useNavigate();
+
+  const wishlsitClickHandler = async () => {
+    try {
+      let res = null;
+      if (!product.wished) {
+        res = await AddToWishlistService(product, token);
+      } else {
+        res = await DeleteFromWishlistService(product._id, token);
+      }
+
+      if (res.status === 200 || res.status === 201) {
+        dispatch({
+          type: actionTypes.SET_WISHLIST,
+          payload: { wishlist: res.data.wishlist },
+        });
+      }
+    } catch (err) {
+      console.log(err);
```

Comment:
You can also go with console.error . Just pointing it out  . Nothing too important